pages/docs/tutorials/native/gradle-for-kotlin-native.md

```diff
@@ -0,0 +1,161 @@
+files are located, which is `src/main/kotlin`. We can of course change this behaviour by defining the `srcFiles`, which we'll see further down. 
+
+We can now create a Kotlin file with the following contents and place it in the `src/main/kotlin` folder:
+
+```kotlin
+fun main(args: Array<String>) {
+    println("Hello Gradle!")
+}
+``` 
+
+To recap, this should be our directory structure layout:
+
+![Directory Structure Before Build]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/dir-before-build.png')}}))
+
+
+From the root directory, where our `build.gradle` file is, we can now run the following command:
+
+    gradle build
+    
+    
+which will build the application. The end result should be the following:
+
+![Directory Structure After Build]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/dir-after-build.png')}}))
+     
+
+We can see that there is a new `build` directory with a subdirectory for each target. Inside this folder we have the actual executable with the name
+we passed to the the parameter `program` in the build script.
+
+## Building and referencing a library
+
+A common process in building applications is to build and reference libraries, be these native libraries we create from scratch in Kotlin, or as a result of interop with C or other languages.
+
+In Gradle we can combine all of this in a single script:
+ 
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
+        maven {
+            url "https://dl.bintray.com/jetbrains/kotlin-native-dependencies"
+        }
+    }
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-native-gradle-plugin:+"
+    }
+}
+
+apply plugin: 'konan'
+
+konan.targets = ['macbook']
+
+konanArtifacts {
+    library('utils') {
+        srcFiles fileTree('src/libs/utils')
+    }
+
+    program('hello') {
+        libraries {
+            artifact 'utils'
+        }
+    }
+}
+```
+
+In this example we're building our library, with the source being in a custom directory and not the default one by convention. We then reference this library 
+during the building of our actual application.
+
+The output structure after running `gradle build` should be:
+
+![Directory Structure After Lib Build]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/dir-lib-build.png')}}))
+
+We can execute our application by referencing it directly or use Gradle via:
+
+    gradle run
+
+    
+resulting in:
+    
+![Gradle Run Result]({{ url_for('tutorial_img', filename='native/gradle-for-kotlin-native/gradle-run.png')}}))    
+
+## Generating CMake files
+
+In order to be able to use CLion, we need to have our project defined using CMake. The Gradle plugins provides us a convenient way to do this
+using the the command `generateCMake`:
+
+    gradle generateCMake
+    
+The result of running this action would give us a new file named `CMakeLists.txt` which allows us to open the project in CLion.
```

Comment:
"The result ... would give" is redundant. "The result .. is a new file" or "Running this action would give us a new file".